sql/src/main/java/io/crate/execution/engine/indexing/GroupRowsByShard.java

```diff
@@ -49,16 +49,22 @@
 
     private final RowShardResolver rowShardResolver;
     private final List<? extends CollectExpression<Row, ?>> expressions;
+    private final List<? extends CollectExpression<Row, ?>> sourceInfoExpressions;
     private final Function<String, TItem> itemFactory;
     private final Supplier<String> indexNameResolver;
     private final ClusterService clusterService;
     private final boolean autoCreateIndices;
+    private final BiConsumer<ShardedRequests, String> itemFailureConsumer;
+    private final Function<ShardedRequests, Boolean> possibleSourceUriProcessor;
 
     GroupRowsByShard(ClusterService clusterService,
                      RowShardResolver rowShardResolver,
                      Supplier<String> indexNameResolver,
                      List<? extends CollectExpression<Row, ?>> expressions,
+                     List<? extends CollectExpression<Row, ?>> sourceInfoExpressions,
                      Function<String, TItem> itemFactory,
+                     BiConsumer<ShardedRequests, String> itemFailureConsumer,
+                     Function<ShardedRequests, Boolean> possibleSourceUriProcessor,
```

Comment:
Personally I prefer to name `Function` variables like you would in a language where functions are first class objects - meaning to give them typical function names. Maybe just `isSourceUriProcessor` ? The `possible` is a bit strange - it implies that if the function returns true it might not be one after all. Is that the case?

In this case I'm wondering why we're not using a `Predicate` - the return value is not nullable as far as I can see.